manta-crypto/src/signature/mod.rs

```diff
@@ -116,6 +121,62 @@ where
 /// Randomness Type
 pub type Randomness<T> = <T as RandomnessType>::Randomness;
 
+/// Signed Message
+#[cfg_attr(
+    feature = "serde",
+    derive(Deserialize, Serialize),
+    serde(crate = "manta_util::serde", deny_unknown_fields)
+)]
+#[derive(derivative::Derivative)]
+#[derivative(
+    Clone(bound = "T::Signature: Clone, T::Message: Clone"),
+    Copy(bound = "T::Signature: Copy, T::Message: Copy"),
+    Debug(bound = "T::Signature: Debug, T::Message: Debug"),
+    Default(bound = "T::Signature: Default, T::Message: Default"),
+    Eq(bound = "T::Signature: Eq, T::Message: Eq"),
+    Hash(bound = "T::Signature: Hash, T::Message: Hash"),
+    PartialEq(bound = "T::Signature: PartialEq, T::Message: PartialEq")
+)]
+pub struct SignedMessage<T>
+where
+    T: MessageType + SignatureType,
+{
+    /// Signature
+    pub signature: T::Signature,
+
+    /// Message
+    pub message: T::Message,
+}
+
+impl<T> SignedMessage<T>
+where
+    T: MessageType + SignatureType,
+{
+    /// Generates a new [`SignedMessage`] by signing `message` with `signing_key`.
+    #[inline]
+    pub fn new<COM>(
+        parameters: &T,
```

Comment:
imo, `parameters` is a signer to sign the `message` and generate a `signature`. Should we create a signer in this function instead of passing it as an argument?

Reply:
We should pass it by reference since the signature parameters can be shared amongst different objects and perhaps cannot be created. Think the same way as Poseidon parameters. We don't want to recreate them every time we use them, instead have one copy and pass around a pointer to it.

Reply:
This `parameter` is the signer of type `T`, which is already generic.